src/nplinker/nplinker.py

```diff
@@ -168,34 +168,50 @@ def scoring_methods(self) -> list[str]:
         """Get names of all valid scoring methods."""
         return list(self._valid_scoring_methods.keys())
 
-    def load_data(self):
-        """Load all data from files into memory.
-
-        This method is a convenience function that calls the
-        [`DatasetArranger`][nplinker.arranger.DatasetArranger] class to arrange data files
-        (download, generate and/or validate data) in the [correct directory structure][working-directory-structure],
-        and then calls the [`DatasetLoader`][nplinker.loader.DatasetLoader] class to load all data
-        from the files into memory.
+    def export_objects(self, objects: Sequence[BGC | Spectrum], filename: str) -> None:
+        """Exports the data for a list of BGC or Spectrum objects to a specified file in tab-separated format.
 
-        The loaded data is stored in various data containers for easy access, e.g.
-        [`self.bgcs`][nplinker.NPLinker.bgcs] for all BGC objects,
-        [`self.strains`][nplinker.NPLinker.strains] for all Strain objects, etc.
+        Args:
+            objects (list[BGC | Spectrum]): A list of BGC or Spectrum objects to be exported.
```

Comment:
GCF and MolecularFamily should also be added to this function, and it's necessary to implement their `to_dict` method too.

Reply:
discussed this with Dora, and we both agreed that it is not necessary to export the GCF and MolecularFamily objects, as they do not contain many attributes directly tied to them.

The most important need is to quickly access which BGCs/Spectra are members of which GCF/MolecularFamily, respectively, and this is easiest by filtering the BGC/Spectrum output TSV files. For the GCF, the only interesting attribute is the BiG-SCAPE class, which I instead included as a column when exporting the BGC objects. The MolecularFamily objects do not even contain any interesting attributes.

Do you think it still makes sense to export a file for the GCFs and MolecularFamilies, perhaps for the sake of completeness? I doubt they will get more attributes in the future, but it’s possible.

Reply:
Ok, we don't have to add it if you believe it's not a need for end users.